### DISCUSSION TITLE
openssl 3.2 can not parse the private pem with passwd encrypt using openssl1.0

### DISCUSSION THREAD

**Discussion by tcjsea** [openssl/openssl#23044] [Q&A] [answered]
When I use openssl-1.0 to create an PKCS#8 private key with passwd encrypt
`openssl pkcs8 -topk8 -inform PEM -in rsa_private_key_PKCS#1.pem -outform pem -out rsa_private_key_PKCS#8_Encrypt_openssl_1.0.pem`

I can parse it with openssl-1.0 cmd as bellow:
`openssl rsa -in rsa_private_key_PKCS#8_Encrypt_openssl_1.0.pem -passin pass:XXXX-out rsa_private.key`

But I can not parse it with openssl-3.2 same cmd:
`openssl-3.2 rsa -in -in rsa_private_key_PKCS#8_Encrypt_openssl_1.0.pem -passin pass:XXXX -out rsa_private.key`
it report bellow error:
```
Could not find private key from rsa_private_key_PKCS#8_encrypt_openssl_1.0.pem
C0214D47117F0000:error:1608010C:STORE routines:ossl_store_handle_load_result:unsupported:crypto/store/store_result.c:151:
C0214D47117F0000:error:0308010C:digital envelope routines:inner_evp_generic_fetch:unsupported:crypto/evp/evp_fetch.c:342:Global default library context, Algorithm (PBKDF1 : 0), Properties ()
```

Can anyone help to find the reason ?

**Comment by t8m** [comment, marked as answer]
It seems it uses PBKDF1 algorithm which is in the legacy provider. Please try:

`openssl-3.2 rsa -provider default -provider legacy -in rsa_private_key_PKCS#8_Encrypt_openssl_1.0.pem -passin pass:XXXX -out rsa_private.key`

**Reply by tcjsea** [reply]
oh, it works, thanks very much for your help ~
